packages/data-service/src/data-service.ts

```diff
@@ -1041,6 +1042,24 @@ class DataServiceImpl extends WithLogContext implements DataService {
    */
   private _unboundLogger?: UnboundDataServiceImplLogger;
 
+  private _getOptionsWithFallbackReadPreference(
+    options?: { readPreference?: ReadPreferenceLike },
+    executionOptions?: { fallbackReadPreference?: ReadPreferenceMode }
+  ) {
+    if (!options || !executionOptions) {
+      return options;
```

Comment:
If options aren't set don't we still want to use the fallback read preference? 

Nice cleanup here and in indexes!

Reply:
Oh, whoops, good catch! Let me add some coverage for this also